src/electionScraper.js

```diff
@@ -0,0 +1,31 @@
+import Election from "./election";
+
+export const INVALID_ELECTION_ID = -1;
+
+export class ElectionScraper {
+
+    /** @type {Map<number, Election>} */
+    elections = new Map();
+
+    /**
+     * @summary extracts election id from election URL
+     * @param {string} electionUrl election URL to extract from
+     * @returns {number}
+     */
+    static getIdFromElectionUrl(electionUrl) {
+        const [, id] = /^https:\/\/(?:www\.)?(?:[^\/]+\.)+(?:com|net|org)\/election\/(\d+)$/.exec(electionUrl) || [];
```

Comment:
if it's just getting the numeric id from the election url, we don't need to validate the full url (not sure if they will start using other TLDs in future).

we could just do something like 
```
/\/election\/(\d+)$/.exec(electionUrl)
```

Reply:
True that - overdid things a little :) After all, you already have this in `validElectionUrl` (we should probably make it `static` as it does not hold any references to the `Election` instance)